### PR TITLE
Return register operation result as int instead of boolean with 3 pos…

### PR DESCRIPTION
**Pull request by aminebagga**
@richirisu could you please review these changes ?

### CODE REVIEW

**Comment by richirisu** [comment]
I am sorry for not having replied earlier. I would like to make some minor suggestions. I would have made these changes myself already, but unfortunately you haven't ticked the box for "allow edits by maintainers" for this pull request. (According to my knowledge it is possible to hit the edit button for this pull request and then change "allow edits by maintainers" afterwards.)

Anyhow, my comments will follow now :)

File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java [hunk 1]
```diff
@@ -93,6 +93,11 @@ public class WifiIotPlugin
   private static final Map<Integer, Result> resultMap = new HashMap<Integer, Result>();
   private static final int ACTIVITY_RESULT_REQUEST_CODE_ADD_NETWORKS = 66778899;
 
+  // Register network reuslts
```
Comment (richirisu) [inline packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java]: fix spelling and wording; for example, "register network result flags"

File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java [hunk 2]
```diff
@@ -258,7 +263,25 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
     if (resultMap.containsKey(requestCode)) {
       final Result result = resultMap.get(requestCode);
       if (requestCode == ACTIVITY_RESULT_REQUEST_CODE_ADD_NETWORKS) {
-        result.success(resultCode == Activity.RESULT_OK);
+        if ((data != null) && data.hasExtra(android.provider.Settings.EXTRA_WIFI_NETWORK_RESULT_LIST)) {
```
Comment (richirisu) [inline packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java]: There might be a problem that a result other than 0, 1, 2 might be returned (error or future addition). In this case the return method would not be triggered. So I would suggest the following:

```dart
int resultValue = -1;
if (data != null && data.hasExtra(EXTRA_WIFI_NETWORK_RESULT_LIST)) {
  for (int code : data.getIntegerArrayListExtra(EXTRA_WIFI_NETWORK_RESULT_LIST)) {
    if (code == ADD_WIFI_RESULT_SUCCESS) {
      resultValue = WIFI_REGISTER_NETWORK_RESULT_SUCCESS;
    } else if (code == ADD_WIFI_RESULT_FAILED) {
      resultValue = WIFI_REGISTER_NETWORK_RESULT_FAILED;
    } else if (code == android.provider.Settings.ADD_WIFI_RESULT_ALREADY_EXISTS) {
      resultValue = WIFI_REGISTER_NETWORK_RESULT_ALREADY_EXISTS;
    }
    break;
  }
}
if (resultCode != -1) {
  result.error("Error", "Unknown return value for ADD_WIFI_RESULT", "");
} else {
  result.success(resultCode);
}
```

One might also import android.provider.Settings

File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java [hunk 3]
```diff
@@ -1020,7 +1042,10 @@ private void registerWifiNetwork(final MethodCall poCall, final Result poResult)
         // notification area
         final int status = moWiFi.addNetworkSuggestions(networkSuggestions);
         Log.e(WifiIotPlugin.class.getSimpleName(), "status: " + status);
-        poResult.success(status == WifiManager.STATUS_NETWORK_SUGGESTIONS_SUCCESS);
+        if (status == WifiManager.STATUS_NETWORK_SUGGESTIONS_SUCCESS)
```
Comment (richirisu) [inline packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java]: Make it inline. (At least I would add curly brackets for the if..else statement.)

```dart
poResult.success(status == WifiManager.STATUS_NETWORK_SUGGESTIONS_SUCCESS ? WIFI_REGISTER_NETWORK_RESULT_SUCCESS : WIFI_REGISTER_NETWORK_RESULT_FAILED);
```

File: packages/wifi_iot/lib/wifi_iot.dart [hunk 4]
```diff
@@ -391,9 +397,11 @@ class WiFiForIoTPlugin {
   ///
   /// @param [isHidden] Whether the SSID is hidden (not broadcasted by the AP).
   ///
-  /// @returns True in case the requested network could be registered, false
-  ///   otherwise.
-  static Future<bool> registerWifiNetwork(
+  /// @returns :
```
Comment (richirisu) [inline packages/wifi_iot/lib/wifi_iot.dart]:
```
@returns Whether the network could be registered, the network failed to register, or the network is already registered.
```

File: packages/wifi_iot/lib/wifi_iot.dart [hunk 5]
```diff
@@ -425,7 +433,13 @@ class WiFiForIoTPlugin {
     } on MissingPluginException catch (e) {
       print("MissingPluginException : ${e.toString()}");
     }
-    return bResult ?? false;
+
```
Comment (richirisu) [inline packages/wifi_iot/lib/wifi_iot.dart:434]: Include "failed" as normal choice AND as fallback.
Add curly brackets.

```
if (iResult == 0) {
  return RegisterWifiNetworkResult.success;
} else if (iResult = 1) {
  return RegisterWifiNetworkResult.failed;
} else if (iResult == 2) {
  return RegisterWifiNetworkResult.alreadyRegistered;
} else {
  return RegisterWifiNetworkResult.failed;
}
```

File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java (hunk 1 above)
Comment (aminebagga) [reply packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java]: Fixed

File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java (hunk 2 above)
Comment (aminebagga) [reply packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java]: Fixed

File: packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java (hunk 3 above)
Comment (aminebagga) [reply packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java]: Fixed

File: packages/wifi_iot/lib/wifi_iot.dart (hunk 4 above)
Comment (aminebagga) [reply packages/wifi_iot/lib/wifi_iot.dart]: Fixed

File: packages/wifi_iot/lib/wifi_iot.dart (hunk 5 above)
Comment (aminebagga) [reply packages/wifi_iot/lib/wifi_iot.dart:434]: Fixed

**Comment by aminebagga** [comment]
> I am sorry for not having replied earlier. I would like to make some minor suggestions. I would have made these changes myself already, but unfortunately you haven't ticked the box for "allow edits by maintainers" for this pull request. (According to my knowledge it is possible to hit the edit button for this pull request and then change "allow edits by maintainers" afterwards.)
> 
> Anyhow, my comments will follow now :)

I did'nt find the "allow edits by maintainers" option, I finished fixed, I let you review and resolve conversion.
I will squash commits once you approved the PR and merge it to your branch

**Comment by richirisu** [comment]
> I did'nt find the "allow edits by maintainers" option, I finished fixed, I let you review and resolve conversion. I will squash commits once you approved the PR and merge it to your branch

I checked and it's now on the right hand side of a pull request, right below the icons of the participants.
Anyway, thanks for making all these changes :)

**Comment by richirisu** [comment]
I run some tests and confirmed on Android 14 that the correct values are all being returned (success, failed, alreadyRegistered).

**Pull request merged**